### PR TITLE
BYOL for AWS and Azure in Mist

### PR DESCRIPTION
**Pull request by Chr1st0ph3rTurn3r**

### CODE REVIEW

File: docs/intro_installation_quickstart_mist_aws.md [hunk 1]
```diff
@@ -5,23 +5,26 @@ sidebar_label: Installing Mist-Managed Router in AWS
...
+
 * Hourly: This provides a free trial period for 30 days and an hourly software cost after the trial expires. This plan is recommended for Proof of Concepts and Trials only. Software upgrades and deployments outside of the cloud, (on premises) require a token or certificate. The software can not be purchased via the marketplace. Refer to the [Session Smart Networking Platform (PAYG)](https://aws.amazon.com/marketplace/pp/prodview-l5kwn7puwvt3g?sr=0-1&ref_=beagle&applicationId=AWSMPContessa) offering.
 
-Once you have selected the AMI that better suits the needs of your deployment, proceed to the [Session Smart Router Deployment](#session-smart-router-deployment) to deploy a Session Smart Router.
+* Bring Your Own License (BYOL): This allows you to install your own licensed copy of the SSR software on an AWS VM. A token or certificate is required to install the software. If a token or certificate is not in your possession, please contact your Juniper Sales representative. Refer to the [Session Smart Networking Platform (BYOL)](https://aws.amazon.com/marketplace/pp/prodview-lz6cjd43qgw3c?sr=0-2&ref_=beagle&applicationId=AWSMPContessa) offering.
```
Comment (MichaelBaj) [inline docs/intro_installation_quickstart_mist_aws.md:23]: No more certificates.

**Review by MichaelBaj — APPROVED** [review]

File: docs/intro_installation_azure_mist.md [hunk 2]
```diff
@@ -93,26 +94,29 @@ The following image shows the infrastructure elements deployed:
...
+4. Click on the tab **Plans + Pricing** as shown in the following picture:
 
   ![Plans](/img/platforms_azure_plans.png)
 
-5. Click the "Launch" link of the Mist-managed template.
+5. Click the **Launch** link of the Mist-managed template.
+
+![CloudFormation Template](/img/azure-byol-template.png)
+
+Answer the following 4 questions to launch the deployment of an SSR. For additional information refer to [Launch the Template](#launch-the-template).
```
Comment (haberkornsam) [inline docs/intro_installation_azure_mist.md:108]: BYOL will have another "question" to answer `What SSR version do you want to install?`

Comment (haberkornsam) [reply docs/intro_installation_azure_mist.md:108]: This goes in the `version` field in the templates

File: docs/intro_installation_azure_mist.md [hunk 3]
```diff
@@ -164,11 +168,11 @@ write_files:
 | Option | Meaning |
 | ------ | ------- |
 | name | The name of the router to use for Mist onboarding. By default, the instance name will be used. |
-| registration-code | The Mist registration used for adoption of the instance to a Mist org. |
+| registration-code | The Mist registration used for adoption of the instance to a Mist organization. |
```
Comment (haberkornsam) [inline docs/intro_installation_azure_mist.md:171]: Need to add `ssr-version` to the onboarding config for BYOL

File: docs/intro_installation_azure_mist.md [hunk 4]
```diff
@@ -265,21 +267,21 @@ A description of the parameters of the template are listed in the following tabl
 | Private Subnet Allowed CIDR    | It corresponds to the source IP CIDR range of the internal workloads/endpoints allowed to originate traffic to the private interface of the router. This field allows for defining a well defined and trusted IP address range. By default is set to 0.0.0.0/0 to allow every workload/endpoint to communicate with the router.                                                                             |
 | Management Subnet Name  | The name of the management subnet within the VNet. |
 | Admin Allowed CIDR      | It allows for restricting reachability to the management interface of the router to a well known source IP address CIDR range. By default is set to 0.0.0.0/0 allowing every IP address to reach the management interface. Once the deployment completes, it is highly recommended to update the configuration of the network security group to allow only access from the source IP address/es where the Session Smart Router will be administered. |
-| Registration Code   | The Mist registration used for adoption of the instance to a Mist org. |
+| Registration Code   | The Mist registration used for adoption of the instance to a Mist organization. |
```
Comment (haberkornsam) [inline docs/intro_installation_azure_mist.md:270]: The template also has `version` which corresponds to `ssr-version`

File: docs/intro_installation_quickstart_mist_aws.md [hunk 5]
```diff
@@ -280,10 +283,10 @@ aws ec2 create-launch-template \
 :::important
 When logging to the Linux instance via SSH use `t128` as the username and the SSH public key of the IAM user provided in the template.
 If a template of the Bring Your Own License image was used, SSH to the EC2 instance using `t128` as the username as indicated in the `SSHLogin` field. Launch the software installation process with the command `sudo install-ssr`.
```
Comment (haberkornsam) [inline docs/intro_installation_quickstart_mist_aws.md:285]: I don't think this applies anymore

**Review by haberkornsam — COMMENTED** [review]
Good start. It might make sense to separate BYOL from the rest of the azure/aws stuff. It is pretty different as far as templates/fields/process goes. Mist managed doesn't use artifactory tokens, but they are required for conductor/conductor-managed

There are also some version restrictions that are at play. i.e. mist > 6.x, conductor > 6.3, etc. And initially only mist-managed is available.

File: docs/intro_installation_byol_azure_mist.md [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+* [Selecting the Azure Plan](#selecting-the-azure-plan).
+* Deploying a [Session Smart Router](#session-smart-router).
+
+## Selecting the Azure Plan
+
+**Bring Your Own License (BYOL):** This allows you to install your own licensed copy of the SSR software on an Azure VM. The device registration code is used to authenticate access to the Mist installation repositories.
+
+Once you have selected the plan that best suits the needs of your deployment, proceed to the [Session Smart Router Deployment](#session-smart-router) to deploy a Session Smart Router.
+
+### Requesting Access to a Private Plan
```
Comment (agrawalkaushik) [inline docs/intro_installation_byol_azure_mist.md:18]: for BYOL there is no need for Private Plan, so we can remove this section

File: docs/intro_installation_byol_azure_mist.md [hunk 7]
```diff
@@ -0,0 +1,365 @@
...
+* What name do you want to give it?
+  * Provide the name in the **Instance Name** field (for example: `128TRouter`).
+* Where do you want to deploy it?
+  * Provide the location where the VNet exists in the **Location** field (for example: eastus). All available locations [here](https://azure.microsoft.com/en-us/global-infrastructure/locations). Note the name of the Location field is one word and all lowercase like eastus, westus, westeurope, eastasia, etc.
+  * Provide the name of the VNet in the **Virtual Network Name** field (for example: `128T-VNet`).
+  * Provide the name of the availability set in the **Availability Set Name** field (for example: `128TRouterSet`).
+  * Provide the name of the **Public Subnet Name**
+  * Provide the name of the **Private Subnet Name**
+  * Provide the name of the **Management Subnet**
+* Which Mist organization is going to manage it?
```
Comment (agrawalkaushik) [inline docs/intro_installation_byol_azure_mist.md:86]: @haberkornsam we need to insert a step here to Provide the SSR version.

File: docs/intro_installation_byol_azure_mist.md [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+Once the deployment completes, information is provided in the Outputs tab on the left hand side.
+
+The non-interactive, Zero Touch Provisioning (ZTP) method is triggered. After the VM is deployed, it will take an additional 2-3 minutes for the ZTP process to complete. When the ZTP process concludes, there will be an asset in the Mist inventory to be associated with the router configuration.
+
+### Cloud-init Onboarding
+When launching an instance using CLI or Powershell or while using automation the following user-data section can be leveraged to setup the onboarding data for the instance.
+
+```
+#cloud-config
+write_files:
```
Comment (agrawalkaushik) [inline docs/intro_installation_byol_azure_mist.md:131]: @haberkornsam same deal here, we should include the SSR version input in this example.

File: docs/intro_installation_byol_azure_mist.md [hunk 9]
```diff
@@ -0,0 +1,365 @@
...
+
+## Troubleshooting
+
+### Validation Process Failure
+
+If the validation process fails with the error shown below, please verify you are deploying the version of the software Juniper has allowlisted to your Subscription ID and try again later. If the problem persists, please contact your Juniper representative.
+
+<img src={useBaseUrl('/img/platforms_azure_private_image_version_mismatch.png')} alt="Private Image Version Mismatch" width="75%" height="75%" />
+
+### Device Does Not Exist In Mist after ZTP
```
Comment (agrawalkaushik) [inline docs/intro_installation_byol_azure_mist.md:164]: @haberkornsam we added a new lite TSI for this case, can you please update the section based on those instructions.

Comment (agrawalkaushik) [reply docs/intro_installation_byol_azure_mist.md:164]: Additionally, we added a command line tool that a user can run to onboard the router (if they do not use cloud-formation/cloud-init etc). So please make sure we capture that as well.

File: static/img/azure-byol-template.png
Comment (agrawalkaushik) [inline static/img/azure-byol-template.png:1]: @haberkornsam we'll need a new screenshot that includes the SSR version

File: static/img/aws-byol-template.png
Comment (agrawalkaushik) [inline static/img/aws-byol-template.png:1]: The graphic is a bit small - but just checking to make sure the SSR version is included here? If not, we need a new screenshot with that - @haberkornsam please help with this

File: static/img/azure-byol-template.png
Comment (haberkornsam) [reply static/img/azure-byol-template.png:1]: The SSR version is included in the screenshot. It's just called `version`. Should we update the templates to say `SSR Version`?

File: docs/intro_installation_byol_azure_mist.md (hunk 6 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/intro_installation_byol_azure_mist.md:18]: removed

File: docs/intro_installation_byol_azure_mist.md (hunk 7 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/intro_installation_byol_azure_mist.md:86]: That step is at the top of the list. I assumed it was the first thing on the template. I probably should have looked more closely at the azure template.

**Review by MichaelBaj — APPROVED** [review]

**Review by agrawalkaushik — APPROVED** [review]

**Review by MichaelBaj — APPROVED** [review]

**Pull request merged**